x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/bulk_actions/bulk_enable_disable_rules.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RulesClient } from '@kbn/alerting-plugin/server';
+import { invariant } from '../../../../../../../common/utils/invariant';
+import type { PromisePoolError } from '../../../../../../utils/promise_pool';
+import type { MlAuthz } from '../../../../../machine_learning/authz';
+import type { RuleAlertType } from '../../../../rule_schema';
+import { validateBulkEnableRule } from '../../../logic/bulk_actions/validations';
+
+interface BulkEnableDisableRulesArgs {
+  rules: RuleAlertType[];
+  action: 'enable' | 'disable';
+  isDryRun?: boolean;
+  rulesClient: RulesClient;
+  mlAuthz: MlAuthz;
+}
+
+interface BulkEnableDisableRulesOutcome {
+  updatedRules: RuleAlertType[];
+  errors: Array<PromisePoolError<RuleAlertType, Error>>;
+}
+
+export const bulkEnableDisableRules = async ({
+  rules,
+  isDryRun,
+  rulesClient,
+  action: operation,
+  mlAuthz,
+}: BulkEnableDisableRulesArgs): Promise<BulkEnableDisableRulesOutcome> => {
+  const errors: Array<PromisePoolError<RuleAlertType, Error>> = [];
+  const updatedRules: RuleAlertType[] = [];
+
+  // In the first step, we validate if the rules can be enabled
+  const validatedRules: RuleAlertType[] = [];
+  await Promise.all(
+    rules.map(async (rule) => {
+      try {
+        await validateBulkEnableRule({ mlAuthz, rule });
+        validatedRules.push(rule);
+      } catch (error) {
+        errors.push({ item: rule, error });
+      }
+    })
+  );
+
+  if (isDryRun || validatedRules.length === 0) {
+    return {
+      updatedRules: validatedRules,
+      errors,
+    };
+  }
+
+  // Then if it's not a dry run, we enable the rules that passed the validation
+  const ruleIds = validatedRules.map(({ id }) => id);
+
+  // Perform actual update using the rulesClient
+  let results;
```

Comment:
nit: `results` doesn't have to a variable. In this case it's necessary only to handle both `enable` and `disable` cases. It could be a const if defined like

```
const results = operation === 'enable' ? await rulesClient.bulkEnableRules({ ids: ruleIds }) : await rulesClient.bulkDisableRules({ ids: ruleIds });
```

or a separate function to return results.